dev/breeze/src/airflow_breeze/shell/enter_shell.py

```diff
@@ -0,0 +1,176 @@
+from airflow_breeze.utils.run_utils import (
+    filter_out_none,
+    instruct_for_setup,
+    md5sum_check_if_build_is_needed,
+    run_command,
+)
+from airflow_breeze.visuals import ASCIIART, ASCIIART_STYLE, CHEATSHEET, CHEATSHEET_STYLE
+
+PARAMS_TO_ENTER_SHELL = {
+    "HOST_USER_ID": "host_user_id",
+    "HOST_GROUP_ID": "host_group_id",
+    "COMPOSE_FILE": "compose_files",
+    "PYTHON_MAJOR_MINOR_VERSION": "python_version",
+    "BACKEND": "backend",
+    "AIRFLOW_VERSION": "airflow_version",
+    "INSTALL_AIRFLOW_VERSION": "install_airflow_version",
+    "AIRFLOW_SOURCES": "airflow_sources",
+    "AIRFLOW_CI_IMAGE": "airflow_ci_image_name",
+    "AIRFLOW_CI_IMAGE_WITH_TAG": "airflow_ci_image_name_with_tag",
+    "AIRFLOW_PROD_IMAGE": "airflow_prod_image_name",
+    "AIRFLOW_IMAGE_KUBERNETES": "airflow_image_kubernetes",
+    "SQLITE_URL": "sqlite_url",
+    "USE_AIRFLOW_VERSION": "use_airflow_version",
+    "SKIP_TWINE_CHECK": "skip_twine_check",
+    "USE_PACKAGES_FROM_DIST": "use_packages_from_dist",
+    "EXECUTOR": "executor",
+    "START_AIRFLOW": "start_airflow",
+    "ENABLED_INTEGRATIONS": "enabled_integrations",
+    "GITHUB_ACTIONS": "github_actions",
+    "ISSUE_ID": "issue_id",
+    "NUM_RUNS": "num_runs",
+    "VERSION_SUFFIX_FOR_SVN": "version_suffix_for_svn",
+    "VERSION_SUFFIX_FOR_PYPI": "version_suffix_for_pypi",
+}
+
+PARAMS_FOR_SHELL_CONSTANTS = {
+    "SSH_PORT": SSH_PORT,
+    "WEBSERVER_HOST_PORT": WEBSERVER_HOST_PORT,
+    "FLOWER_HOST_PORT": FLOWER_HOST_PORT,
+    "REDIS_HOST_PORT": REDIS_HOST_PORT,
+    "MYSQL_HOST_PORT": MYSQL_HOST_PORT,
+    "MYSQL_VERSION": MYSQL_VERSION,
+    "MSSQL_HOST_PORT": MSSQL_HOST_PORT,
+    "MSSQL_VERSION": MSSQL_VERSION,
+    "POSTGRES_HOST_PORT": POSTGRES_HOST_PORT,
+    "POSTGRES_VERSION": POSTGRES_VERSION,
+}
+
+PARAMS_IN_CACHE = {
+    'python_version': 'PYTHON_MAJOR_MINOR_VERSION',
+    'backend': 'BACKEND',
+    'executor': 'EXECUTOR',
+    'postgres_version': 'POSTGRES_VERSION',
+    'mysql_version': 'MYSQL_VERSION',
+    'mssql_version': 'MSSQL_VERSION',
+}
+
+DEFAULT_VALUES_FOR_PARAM = {
+    'python_version': 'DEFAULT_PYTHON_MAJOR_MINOR_VERSION',
+    'backend': 'DEFAULT_BACKEND',
+    'executor': 'DEFAULT_EXECUTOR',
+    'postgres_version': 'POSTGRES_VERSION',
+    'mysql_version': 'MYSQL_VERSION',
+    'mssql_version': 'MSSQL_VERSION',
+}
+
+
+def construct_arguments_docker_compose_command(shell_params: ShellBuilder) -> List[str]:
+    args_command = []
+    for param_name in PARAMS_TO_ENTER_SHELL:
+        param_value = PARAMS_TO_ENTER_SHELL[param_name]
+        args_command.append("-e")
+        args_command.append(param_name + '=' + str(getattr(shell_params, param_value)))
+    for constant_param_name in PARAMS_FOR_SHELL_CONSTANTS:
+        constant_param_value = PARAMS_FOR_SHELL_CONSTANTS[constant_param_name]
+        args_command.append("-e")
+        args_command.append(constant_param_name + '=' + str(constant_param_value))
+    return args_command
+
+
```

Comment:
The problem is that `docker-compose up` does not handle `-e` flag :( 

See here:

https://stackoverflow.com/questions/49293967/how-to-pass-environment-variable-to-docker-compose-up

I think you should continue passing them via os.environ (this is basically what is suggested in the last answer as second option.

However I think that requires a bit of modification of the "--verbose" run_command behaviour - you shoudl be able to pass it the list of environment variables to set and the "verbose" information should include it in the command as evn variable setting, so that you can copy&paste it from there.

For example if you run docker-compose with "VAR1"="VAL1" variable, the verbose command should look like:


```
VAR1=VAL1 docker-compose up ......
```

Reply:
@potiuk Do you think it's better to log all the env variable used during this docker-compose command?

Reply:
I think so. The idea for the "verbose" part is to give the user a way to copy&paste the command and re-run it easily (maybe with small modifications) and to be able to diagnose potential bugs in setting the configuration in the CI. 

So logging all variables there (i.e. being able to reproduce the exact command that caused the problem) is very important.